File: RefinerySystem/Refinery/src/models/objects/RechargeRequest.ts
import { FuelType } from "./FuelType";
import { Route } from "./Route";
import { Station } from "./Station";

export interface RechargeRequest {
    id?: number;
    fuel_type: FuelType;
    fuel_type_id: number;
    fuel_quantity: number;
    station_id: number;
    station: Station;
    completed?: boolean;
    route?: Route;
    route_id?: number;
    deliver_order?: number;
}

File: RefinerySystem/RefineryAPI/refinery/models/recharge_request.py
from django.db import models


class RechargeRequest(models.Model):
    fuel_quantity = models.FloatField()
    station_id = models.IntegerField()
    fuel_type = models.ForeignKey('FuelType', on_delete=models.CASCADE)
    route = models.ForeignKey('Route', on_delete=models.SET_NULL, null=True, blank=True,
                              related_name='recharge_requests')
    completed = models.BooleanField(default=False)
    deliver_order = models.IntegerField(null=True, blank=True)


